modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java

```diff
@@ -390,8 +390,19 @@ public void revert(long previousLevel) {
         // walk back removing blocks greater than the given level value
         IAionBlock bestLine = bestBlock;
         while (currentLevel > previousLevel) {
+
+            // remove all the blocks at that level
+            List<BlockInfo> currentLevelBlocks = getBlockInfoForLevel(currentLevel);
+            if (currentLevelBlocks.size() == 0){
+                blocks.delete(bestLine.getHash());
+            } else {
+                for (BlockInfo bk_info : currentLevelBlocks) {
+                    blocks.delete(bk_info.getHash());
+                }
+            }
+
+            // remove the level
             index.remove((int) currentLevel);
-            blocks.delete(bestLine.getHash());
             bestLine = getBlockByHash(bestLine.getParentHash());
             --currentLevel;
```

Comment:
Should we start delete all levels from the max number to the revert number, level by level?

Reply:
`getBestBlock()` uses this method to determine the height:
https://github.com/aionnetwork/aion/blob/dev/modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java#L257-L261
so it will delete all levels higher than the given parameter